concordium-sdk/src/main/java/com/concordium/sdk/responses/blocksummary/updates/Fraction.java

```diff
@@ -0,0 +1,40 @@
+package com.concordium.sdk.responses.blocksummary.updates;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.val;
+
+import java.math.BigInteger;
+
+/**
+ * A fraction
+ */
+@EqualsAndHashCode
+@Getter
+public class Fraction {
+    private final BigInteger denominator;
+    private final BigInteger numerator;
+
+    @JsonCreator
+    public Fraction(@JsonProperty("denominator") BigInteger denominator, @JsonProperty("numerator") BigInteger numerator) {
+        if (denominator.equals(BigInteger.ZERO)) {
+            throw new IllegalArgumentException("Unable to compute gcd.");
+        }
+        val gcd = numerator.gcd(denominator);
+        if (gcd.equals(BigInteger.ZERO)) {
+            this.denominator = denominator;
```

Comment:
Shouldn't we normalize to the fraction (0 / 1) in the case the numerator is 0?

Reply:
Lets do it.